scripts/generate/public_git.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# Copyright (c) F5, Inc.
+# This source code is licensed under the Apache License, Version 2.0 license found in the
+# LICENSE file in the root directory of this source tree.
+
+# shellcheck shell=bash
+# Generate support file from a public git repo
+# Call it through //go:generate comments in project root to update .gen.go files
+
+set -e
+
+branch="master"
+directive_map=""
+match_func=""
+url=""
+path=""
+
+help() {
+    cat << EOF
+Generate support file from a git repo.
+
+usage: $(basename "$0") [-d|--directive-map] [-b|--branch] [--url] [-m|--match-func] [-o|--override] [-p|--path] [h|--help]
+    -h  | --help               Display this message
+    -d  | --directive-map      Name of the generated directive map(required)
+    -b  | --branch             Branch to checkout, defaults to "$branch"(optional)
+    -m  | --match-func         Name of the generated MatchFunc(required)
+    -p  | --path               The path of the output file(required)
+    --url                      Url used for git clone(required)
+EOF
+}
+
+while [ ! $# -eq 0 ]; do
+    case "$1" in
+        --help | -h)
+            help
+            exit 0
+            ;;
+        --directive-map | -d)
+            directive_map="$2"
+            shift
+            ;;
+        --branch | -b)
+            branch="$2"
+            shift
+            ;;
+        --match-func | -m)
+            match_func="$2"
+            shift
+            ;;
+        --url)
+            url="$2"
+            shift
+            ;;
+        --path | -p)
+            path="$2"
+            shift
+            ;;
+        *)
+            printf "Invalid option: %s\n" "$1" 1>&2
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+if [ "$url" = "" ]; then
+    echo "url can't be empty"
+    exit 1
+fi
+
+if [ "$path" = "" ]; then
+    echo "path can't be empty"
+    exit 1
+fi
+
+if [ "$directive_map" = "" ]; then
+    echo "directive-map can't be empty"
+    exit 1
+fi
+
+if [ "$match_func" = "" ]; then
+    echo "match-func can't be empty"
+    exit 1
+fi
+
+echo "start generate from $url"
```

Comment:
Nit: "No news is good news". I think don't print anything unless asked to do so. You may even wish to print the results to `stdout` and let the caller redirect to a file so you don't need the require the `--path` option? The appearance is more or less similar in the `//go:generate` directives but requires a little less shell script.

Reply:
> Nit: "No news is good news".

I see! Will change.

> You may even wish to print the results to stdout and let the caller redirect to a file so you don't need the require the --path option?

My caution for it is: when the script runs into error, we will still redirect into files. Not sure if it is acceptable

Reply:
Updated: remove unnecessary echo output